### PR TITLE
Add test plan for QUARKUS-278: RH SSO 7.4

### PR DESCRIPTION
**Pull request by Sgitario**
@rsvoboda this is the proposal test plan. I guess the next is to identify what scenarios will be out of scope for 7.6 testing.

### CODE REVIEW

File: QUARKUS-278.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+## Automated test development
+This step should ensure:
+ - Focus on high-level scenarios with multiple components and features involved
+ - OpenShift should be the primary target platform for product to ensure good interoperability
+ - Ensure platform specific features or components have appropriate test coverage
+
+## Advanced topics for test development
+The following advanced topics are listed for future consideration. There should be dedicated feature requirements and/or enhancements for these specific areas.
+ - Integration testing with other components (gRPC, WebServers, RestClient, ...)
```
Comment (Sgitario) [inline QUARKUS-278.md:51]: @rsvoboda Does this make sense to you?

File: QUARKUS-278.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+### Impact on testsuites and testing automation:
+ - OpenShift TS framework needs to be enhanced to install RH-SSO 7.4 and/or other versions
+ - New test coverage for security scenarios needs to be added into OpenShift TS:
+   - Protect Services Applications with Roles
+   - Protect Services Applications with Roles using permissions managed by RH SSO
+   - Protect Web Applications with Roles
+   - Protect Multi Tenant applications
+   - Protect Service Applications using JWT
+   - Cover Expiration of Bearer Token for Services Applications
+   - User Revocation for Services Applications
```
Comment (Sgitario) [inline QUARKUS-278.md:25]: @rsvoboda these are all the scenarios I've covered so far, but some of them are not even merged into master yet. Therefore, pending to confirm which scenarios will be part of 1.7 coverage.

File: QUARKUS-278.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+## Scope of the testing
+Following actions were taken to ensure test coverage and automation for native image:
+ - Define the scope of the testing
+ - Analyze the possible impact on Quarkus itself - e.g performance, compatibility
+ - Determine impact on test suites - e.g. new test suite, increased complexity of existing test suite, maintenance burden
+ - Determine impact on QE infrastructure - e.g. increased complexity of existing automation, maintenance burden, need for new external resources
+ - Check a need for involvement of other product/component teams
+
+### Impact on testsuites and testing automation:
```
Comment (Sgitario) [inline QUARKUS-278.md:16]: Also, I've added some scenarios to cover more integrations in the beefy project like for example with OIDC and Rest Client: https://github.com/quarkus-qe/beefy-scenarios/pull/54
But I'm not sure if we should or if it's worthy to be noted here.

File: QUARKUS-278.md (hunk 2 above)
Comment (rsvoboda) [reply QUARKUS-278.md:25]: Please mention also existing tests in OpenShift TS. 

And add a note similar to this text:
Quarkus 1.7 focused test development targets reasonable test coverage baseline.
Future test development targets Quarkus 1.11 to gain iteratively stronger confidence.

File: QUARKUS-278.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+   - Protect Services Applications with Roles using permissions managed by RH SSO
+   - Protect Web Applications with Roles
+   - Protect Multi Tenant applications
+   - Protect Service Applications using JWT
+   - Cover Expiration of Bearer Token for Services Applications
+   - User Revocation for Services Applications
+ - Ensure this coverage works both in JVM and NATIVE mode
+
+### Impact on resources:
+ - More services will be running on OpenShift cluster, impact on CPU/memory should be minimal
```
Comment (rsvoboda) [inline QUARKUS-278.md:29]:
```suggestion
 - More services will be running on OpenShift cluster, impact on CPU/memory should be minimal
 - Test execution will be prolonged, preliminarily estimation is ~30 minutes per each OCP stream
```

File: QUARKUS-278.md (hunk 1 above)
Comment (rsvoboda) [reply QUARKUS-278.md:51]: What should I imagine under `WebServers`?

File: QUARKUS-278.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+## Automated test development
+This step should ensure:
+ - Focus on high-level scenarios with multiple components and features involved
+ - OpenShift should be the primary target platform for product to ensure good interoperability
+ - Ensure platform specific features or components have appropriate test coverage
+
+## Advanced topics for test development
+The following advanced topics are listed for future consideration. There should be dedicated feature requirements and/or enhancements for these specific areas.
+ - Integration testing with other components (gRPC, WebServers, RestClient, ...)
+ - Recovery/Failover testing
```
Comment (rsvoboda) [inline QUARKUS-278.md:52]: What would be the scope?

I tend to think this should be in "Out of the scope" section.
This area should be covered by RH-SSO team. Please provide ideas what could be covered under this.

File: QUARKUS-278.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+ - Performance testing
```
Comment (rsvoboda) [inline QUARKUS-278.md:53]: What would be the scope?

I tend to think this should be in "Out of the scope" section.

File: QUARKUS-278.md (hunk 2 above)
Comment (Sgitario) [reply QUARKUS-278.md:25]: The existing tests are the first two scenarios. I will mention it.

File: QUARKUS-278.md (hunk 1 above)
Comment (Sgitario) [reply QUARKUS-278.md:52]: Sorry, I meant WebSockets. Will change it.

File: QUARKUS-278.md (hunk 3 above)
Comment (Sgitario) [reply QUARKUS-278.md:16]: Added

File: QUARKUS-278.md (hunk 2 above)
Comment (Sgitario) [reply QUARKUS-278.md:25]: Changed

File: QUARKUS-278.md (hunk 1 above)
Comment (Sgitario) [reply QUARKUS-278.md:52]: Fixed

File: QUARKUS-278.md (hunk 5 above)
Comment (Sgitario) [reply QUARKUS-278.md:53]: Removed recovery/failover testing.

File: QUARKUS-278.md (hunk 6 above)
Comment (Sgitario) [reply QUARKUS-278.md:54]: Removed performance testing from here.

**Comment by Sgitario** [comment]
@rsvoboda I think I've already addressed all your comments and rewrite the "Impact on testsuites and testing automation:" section.

**Review by rsvoboda — APPROVED** [review]

**Pull request merged**